### PR TITLE
Fix #2960 by checking if the values are a list of lists.

### LINKED ISSUE #2960: [Bug]: Chromadb 0.5.1+ breaks retrieval agent
### Describe the bug

In https://github.com/chroma-core/chroma/pull/2044 a new field was added in the query result `included` which is an array of strings (not None and not [[]]), so when https://github.com/microsoft/autogen/blob/main/autogen/agentchat/contrib/vectordb/utils.py#L110 tries to data_dict[key][i][j], it dies.

### Steps to reproduce

_No response_

### Model Used

Any

### Expected Behavior

Should skip included key or otherwise protect against it.

### Screenshots and logs

```
models to use:  ['gpt-4']
Accepted file formats for `docs_path`:
['txt', 'json', 'csv', 'tsv', 'md', 'html', 'htm', 'rtf', 'rst', 'jsonl', 'log', 'xml', 'yaml', 'yml', 'pdf']
/Users/jpr/.asdf/installs/python/3.12.4/lib/python3.12/site-packages/huggingface_hub/file_download.py:1132: FutureWarning: `resume_download` is deprecated and will be removed in version 1.0.0. Downloads always resume when possible. If you want to force a new download, use `force_download=True`.
  warnings.warn(
Trying to create collection.
2024-06-18 12:20:28,983 - autogen.agentchat.contrib.retrieve_user_proxy_agent - INFO - Use the existing collection `autogen-docs`.
2024-06-18 12:20:29,153 - autogen.agentchat.contrib.retrieve_user_proxy_agent - INFO - Found 146 chunks.
Traceback (most recent call last):
  File "/Users/jpr/src/AITools/autogen/rhino.py", line 58, in <module>
    chat_result = ragproxyagent.initiate_chat(
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/jpr/.asdf/installs/python/3.12.4/lib/python3.12/site-packages/autogen/agentchat/conversable_agent.py", line 1015, in initiate_chat
    msg2send = message(_chat_info["sender"], _chat_info["recipient"], kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/jpr/.asdf/installs/python/3.12.4/lib/python3.12/site-packages/autogen/agentchat/contrib/retrieve_user_proxy_agent.py", line 631, in message_generator
    sender.retrieve_docs(problem, n_results, search_string)
  File "/Users/jpr/.asdf/installs/python/3.12.4/lib/python3.12/site-packages/autogen/agentchat/contrib/retrieve_user_proxy_agent.py", line 562, in retrieve_docs
    results = self._vector_db.retrieve_docs(
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/jpr/.asdf/installs/python/3.12.4/lib/python3.12/site-packages/autogen/agentchat/contrib/vectordb/chromadb.py", line 260, in retrieve_docs
    results = chroma_results_to_query_results(results)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/jpr/.asdf/installs/python/3.12.4/lib/python3.12/site-packages/autogen/agentchat/contrib/vectordb/utils.py", line 111, in chroma_results_to_query_results
    sub_dict[key[:-1]] = data_dict[key][i][j]  # remove 's' in the end from key
                         ~~~~~~~~~~~~~~~~~^^^
IndexError: string index out of range
```

### Additional Information

Downgrading to 0.5.0 works

### PR DESCRIPTION
**Pull request by thinkall**
## Why are these changes needed?

To fix #2960 

## Related issue number

Closes #2960 

## Checks

- [ ] I've included any doc changes needed for https://microsoft.github.io/autogen/. See https://microsoft.github.io/autogen/docs/Contribute#documentation to build and test documentation locally.
- [ ] I've added tests (if relevant) corresponding to the changes introduced in this PR.
- [ ] I've made sure all auto checks have passed.

### CODE REVIEW

**Review by Hk669 — APPROVED** [review]
i think this should handle the out of bounds error, looks good to me. thanks @thinkall

File: autogen/agentchat/contrib/vectordb/utils.py [hunk 1]
```diff
@@ -99,15 +99,19 @@ def chroma_results_to_query_results(data_dict: Dict[str, List[List[Any]]], speci
...
 
-    keys = [key for key in data_dict if key != special_key]
+    keys = [
+        key
+        for key in data_dict
+        if key != special_key and data_dict[key] is not None and isinstance(data_dict[key][0], list)
+    ]
     result = []
 
     for i in range(len(data_dict[special_key])):
```
Comment (Hk669) [inline autogen/agentchat/contrib/vectordb/utils.py:109]:
```suggestion
    for i in range(len(special_keys)):
```

File: autogen/agentchat/contrib/vectordb/utils.py [hunk 2]
```diff
@@ -99,15 +99,19 @@ def chroma_results_to_query_results(data_dict: Dict[str, List[List[Any]]], speci
...
         sub_result = []
         for j, distance in enumerate(data_dict[special_key][i]):
```
Comment (Hk669) [inline autogen/agentchat/contrib/vectordb/utils.py:111]:
```suggestion
        for j, distance in enumerate(special_keys[i]):

**Comment by thinkall** [comment]
> i think this should handle the out of bounds error, looks good to me. thanks @thinkall

Thank you @Hk669 , I've updated code to reduce dictionary look up as you suggested.

**Comment by Hk669** [comment]
> Thank you @Hk669 , I've updated code to reduce dictionary look up as you suggested.

Thanks @thinkall , looks great.

**Pull request merged**